.github/workflows/pr_check_workflow.yml

```diff
@@ -138,76 +66,39 @@ jobs:
     steps:
       - run: echo Running functional tests for ciGroup${{ matrix.group }}
 
-      # Access a cache of set results from a previous run of the job
-      # This is to prevent re-running a CI group that was already successful since it is not native to github actions
-      # Can be used to verify flaky steps with reduced times
-      - name: Restore the cached run
-        uses: actions/cache@v2
-        with:
-          path: |
-            ftr_tests_results
-          key: ${{ github.run_id }}-${{ github.job }}-${{ matrix.group }}-${{ github.sha }}
-          restore-keys: |
-            ${{ github.run_id }}-${{ github.job }}-${{ matrix.group }}-${{ github.sha }}
-
       - name: Get the cached tests results 
         id: ftr_tests_results
         run: cat ftr_tests_results 2>/dev/null || echo 'default'
 
       - name: Checkout code
-        if: steps.ftr_tests_results.outputs.ftr_tests_results != 'success'
         uses: actions/checkout@v2
 
       - name: Setup Node
-        if: steps.ftr_tests_results.outputs.ftr_tests_results != 'success'
         uses: actions/setup-node@v2
         with:
           node-version-file: ".nvmrc"
           registry-url: 'https://registry.npmjs.org'
 
       - name: Setup Yarn
-        if: steps.ftr_tests_results.outputs.ftr_tests_results != 'success'
         run: |
           npm uninstall -g yarn
           npm i -g yarn@1.22.10
 
-      - name: Get cache path
```

Comment:
NIT: I think we should keep `Get cache path` and `Setup cache` for functional tests. but we can remove the if check for success. The reason being that this cache actually is the node_modules cache so it's just a slight optimization since it's not a clean bootstrap for each suite. So if `ciGroup1` builds completes before `ciGroup12` begins to build it will load the node_modules built by `ciGroup1`. But I say nitpick because eyeballing it, it seems like all the ciGroups might start building around the same time

Reply:
Yeah from what I can tell they all run at almost the same time. So using that cache would require one of the groups to have finished bootstrapping before it started, right? I'm not sure that added complexity is worth the not-guaranteed potential of saving ~3 minutes from the execution time. 

Or is there something else I'm missing?